mlptrain/potentials/mace/mace.py

```diff
@@ -0,0 +1,583 @@
+        logger=logger,
+        patience=args.patience,
+        device=device,
+        swa=swa,
+        ema=ema,
+        max_grad_norm=args.clip_grad,
+        log_errors=error_table,
+        )
+
+        epoch = checkpoint_handler.load_latest(
+        state=tools.CheckpointState(model, optimizer, lr_scheduler), device=device
+        )
+        logging.info(f"Loaded model from epoch {epoch}")
+
+        # Evaluation on test datasets
+        logging.info("Computing metrics for training, validation, and test sets")
+        
+        all_collections = [
+        ("train", train_configs),
+        ("valid", valid_configs),
+        ]
+
+        table = self.create_error_table(
+        error_table,
+        all_collections,
+        z_table,
+        r_max,
+        args.valid_batch_size,
+        model,
+        loss_fn,
+        device)
+        # Save entire model
+        model_path = os.path.join(args.checkpoints_dir, tag + ".model")
+
+        logging.info("\n" + str(table))
+
+        logging.info(f"Saving model to {model_path}")
+        if save_cpu:
+            model = model.to("cpu")
+        torch.save(model, model_path)
+
+        logging.info("Done")
+
+        return None
+    
+    def _train(self, **kwargs):
+
+        n_cores = kwargs['n_cores'] if 'n_cores' in kwargs else min(Config.n_cores, 8)
+        os.environ['OMP_NUM_THREADS'] = str(n_cores)
+        logger.info(f'Using {n_cores} cores to train MACE potential')
+
+        for config in self.training_data:
+            if self.requires_non_zero_box_size and config.box is None:
+                config.box = Box([100, 100, 100])
+
+        self.training_data.save_xyz(filename=f'{self.name}_data.xyz')
+        start_time = time()
+        self._train_file()
+        delta_time = time() - start_time
+        logger.info(f'MACE training ran in {delta_time / 60:.1f} m')
+
+        return None
+
+    def get_dataset_from_xyz(self,  train_path, valid_path, valid_fraction, config_type_weights, seed, energy_key, forces_key):
+        """Load training and test dataset from xyz file"""
+        
+        try:
+            from mace import data
+
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError('MACE install not found, install it '
+                                      'here: https://github.com/ACEsuit/mace')
+            
+        _, all_train_configs = data.load_from_xyz(file_path=train_path,
+                                                  config_type_weights=config_type_weights,
+                                                  energy_key=energy_key,
+                                                  forces_key=forces_key,
+                                                  extract_atomic_energies=False)
+        logging.info(f"Loaded {len(all_train_configs)} training configurations from '{train_path}'")
+
+        if valid_path is not None:
+            _, valid_configs = data.load_from_xyz(
+            file_path=valid_path,
```

Comment:
Indentation looks a bit weird to me, difficult to see these args are for load_from_xyz